### PR TITLE
go_1_20: remove; unpin depending packages, fix or mark broken

### PR DESCRIPTION
**Pull request by katexochen**
## Description of changes

Go 1.20 is [EOL](https://endoflife.date/go) since 2024-02-06.

This PR requires:

- [x] https://github.com/NixOS/nixpkgs/pull/294783
- [x] https://github.com/NixOS/nixpkgs/pull/294791

Opens:

- [x] `envoy`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/misc/v2ray-geoip/default.nix [hunk 1]
```diff
@@ -8,17 +9,24 @@
 }:
 
 let
-  generator = pkgsBuildBuild.buildGo120Module {
+  generator = pkgsBuildBuild.buildGoModule rec {
     pname = "v2ray-geoip";
-    version = "unstable-2023-10-11";
+    version = "202403070031";
```
Comment (SuperSandro2000) [inline pkgs/data/misc/v2ray-geoip/default.nix:14]: Or we just update to https://github.com/v2fly/geoip/releases/tag/202403140037

File: pkgs/data/misc/v2ray-geoip/default.nix [hunk 2]
```diff
@@ -8,17 +9,24 @@
...
       repo = "geoip";
-      rev = "3182dda7b38c900f28505b91a44b09ec486e6f36";
-      hash = "sha256-KSRgof78jScwnUeMtryj34J0mBsM/x9hFE4H9WtZUuM=";
+      rev = version;
+      hash = "sha256-xws/qLVrqfyDKcfwoUk4Lbpjj7gcpBNpMOSTQCDf4h8=";
     };
 
+    patches = [
+      (fetchpatch {
+        url = "https://patch-diff.githubusercontent.com/raw/v2fly/geoip/pull/88.patch";
```
Comment (SuperSandro2000) [inline pkgs/data/misc/v2ray-geoip/default.nix:25]:
```suggestion
        url = "https://github.com/v2fly/geoip/commit/dfab705de96c660561454fb2e92abb7af06e8842.patch";
```

**Comment by katexochen** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>go-libp2p-daemon</li>
    <li>magnetico</li>
    <li>norouter</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>dcrwallet</li>
    <li>envoy</li>
    <li>goconvey</li>
    <li>gomacro</li>
    <li>gost</li>
    <li>k3s_1_24</li>
    <li>k3s_1_25</li>
    <li>k3s_1_26</li>
    <li>k3s_1_27</li>
    <li>k3s_1_28</li>
    <li>labctl</li>
    <li>pomerium</li>
    <li>qv2ray</li>
    <li>sing-geosite</li>
    <li>trezor-suite</li>
    <li>trezord</li>
    <li>v2ray</li>
    <li>v2ray-geoip</li>
    <li>v2raya</li>
    <li>xray</li>
    <li>zincsearch</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -32581,17 +32555,9 @@ with pkgs;
...
-    buildGoModule = buildGo120Module;
-  };
-  inherit (callPackage ../applications/networking/cluster/k3s {
-    buildGoModule = buildGo120Module;
-  }) k3s_1_26 k3s_1_27 k3s_1_28;
-  inherit (callPackage ../applications/networking/cluster/k3s { }) k3s_1_29;
-
+  k3s_1_24 = callPackage ../applications/networking/cluster/k3s/1_24 { };
+  k3s_1_25 = callPackage ../applications/networking/cluster/k3s/1_25 { };
+  inherit (callPackage ../applications/networking/cluster/k3s { }) k3s_1_29 k3s_1_26 k3s_1_27 k3s_1_28;
```
Comment (superherointj) [inline pkgs/top-level/all-packages.nix:32560]:
```suggestion
  inherit (callPackage ../applications/networking/cluster/k3s {
    buildGoModule = buildGo121Module;
  }) k3s_1_26 k3s_1_27 k3s_1_28 k3s_1_29;
```

Comment (superherointj) [reply pkgs/top-level/all-packages.nix:32560]: As I had to bump K3s 1.26, 1.27 to use the newer go compiler version.

Opened a PR fixing it here: https://github.com/NixOS/nixpkgs/pull/296155

No need to make changes to K3s just a rebase after PR is merged.

**Comment by SuperSandro2000** [comment]
Can you rebase on master?

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**